packages/app/src/app/pages/Pro/index.tsx

```diff
@@ -268,4 +284,266 @@ const Expiring = ({
   </MaxWidth>
 );
 
+const prettyPermissions = {
+  ADMIN: 'Admin',
+  WRITE: 'Editor',
+  READ: 'Viewer',
+};
+
+const Upgrade = () => {
+  const {
+    state: { personalWorkspaceId, user, activeTeam, activeTeamInfo, dashboard },
+    actions: { setActiveTeam },
+  } = useOvermind();
+
+  const location = useLocation();
+  const history = useHistory();
+  const workspaceId = new URLSearchParams(location.search).get('workspace');
+
+  React.useEffect(() => {
+    // set workspace in url when coming from places without workspace context
+    if (activeTeam && !workspaceId) {
+      history.replace({
+        pathname: location.pathname,
+        search: '?v=2&workspace=' + activeTeam,
+      });
+    }
+  }, [workspaceId, activeTeam, history, location]);
+
+  if (!activeTeam || !dashboard.teams.length) return null;
+
+  const personalWorkspace = dashboard.teams.find(
+    t => t.id === personalWorkspaceId
+  )!;
+
+  const getUserAuthorization = workspace => {
+    const userAuthorization = workspace.userAuthorizations.find(
+      authorization => authorization.userId === user.id
+    ).authorization;
+    return userAuthorization;
+  };
+
+  const workspaces = [
+    personalWorkspace,
+    ...sortBy(
+      dashboard.teams.filter(t => t.id !== personalWorkspaceId),
+      [getUserAuthorization, 'name']
+    ),
+  ];
```

Comment:
The sorting logic here is hiding too much logic, not sure if this is the best way to do it. Would you do it differently?